tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/fragment/FragmentProjectNameValidationService.java

```diff
@@ -15,135 +15,51 @@
-                {
-                    refresh();
-                }
-            }
-        };
-
-        op().attach( listener, "*" );
+        return context( NewModuleFragmentOp.class );
     }
 
     @Override
-    protected Status compute()
+    protected NewLiferayProjectProvider<NewModuleFragmentOp> getProjectProvider()
     {
-        Status retval = Status.createOkStatus();
-
-        final NewModuleFragmentOp op = op();
-        final String currentProjectName = op.getProjectName().content();
-
-        if( !CoreUtil.empty( currentProjectName ) )
-        {
-            final IStatus nameStatus = CoreUtil.getWorkspace().validateName( currentProjectName, IResource.PROJECT );
-
-            if( !nameStatus.isOK() )
-            {
-                return StatusBridge.create( nameStatus );
-            }
-
-            if( ValidationUtil.isExistingProjectName( currentProjectName ) )
-            {
-                return Status.createErrorStatus( "A project with that name(ignore case) already exists." );
-            }
-
-            if( !isValidProjectName( currentProjectName ) )
-            {
-                return Status.createErrorStatus( "The project name is invalid." );
-            }
-
-            final Path currentProjectLocation = op.getLocation().content( true );
-
-            // double check to make sure this project wont overlap with existing dir
-            if( currentProjectLocation != null )
-            {
-                final String currentPath = currentProjectLocation.toOSString();
-                final IPath osPath = org.eclipse.core.runtime.Path.fromOSString( currentPath );
-
-                final IStatus projectStatus =
-                    op.getProjectProvider().content().validateProjectLocation( currentProjectName, osPath );
-
-                if( !projectStatus.isOK() )
-                {
-                    return StatusBridge.create( projectStatus );
-                }
-
-                File projectFodler = osPath.append( currentProjectName ).toFile();
-
-                if( projectFodler.exists() && projectFodler.list().length > 0 )
-                {
-                    return StatusBridge.create(
-                        ProjectCore.createErrorStatus( "Target project folder is not empty." ) );
-                }
-            }
-        }
-
-        return retval;
+        return op().getProjectProvider().content();
     }
 
     @Override
-    public void dispose()
+    protected ValueProperty getProjectNameValueProperty()
     {
-        if( this.listener != null && op() != null && !op().disposed() )
-        {
-            op().detach( listener, "*" );
-
-            this.listener = null;
-        }
-
-        super.dispose();
+        return NewModuleFragmentOp.PROP_PROJECT_NAME;
     }
 
-    private boolean isValidProjectName( String currentProjectName )
+    @Override
+    protected <A> A getData( String key, Class<A> type )
```

Comment:
This looks incorrect.  We shouldn't override getData and then have to hard-code strings for "projectName" or "Location"  that will break if we ever refactored those properties